thoth/user_api/api_v1.py

```diff
@@ -80,9 +81,60 @@
 
 _PROVENANCE_CHECK_PROTECTED_FIELDS = frozenset({"kebechet_metadata"})
 
+_METRIC_CACHE_HIT_ADVISER_AUTHENTICATED = metrics.counter(
+    "thoth_user_api_cache_hit_rate",
+    "Thoth User API cache hit rate",
+    is_authenticated="True",
+    service="adviser",
+    env=Configuration.THOTH_DEPLOYMENT_NAME,
+)
+
+_METRIC_CACHE_HIT_ADVISER_UNHAUTHENTICATED = metrics.counter(
+    "thoth_user_api_cache_hit_rate",
+    "Thoth User API cache hit rate",
+    is_authenticated="False",
+    service="adviser",
+    env=Configuration.THOTH_DEPLOYMENT_NAME,
+)
+
+_METRIC_CACHE_HIT_PROVENANCE_CHECKER_AUTHENTICATED = metrics.counter(
+    "thoth_user_api_cache_hit_rate",
+    "Thoth User API cache hit rate",
+    is_authenticated="True",
+    service="provenance-checker",
+    env=Configuration.THOTH_DEPLOYMENT_NAME,
+)
+
+_METRIC_CACHE_HIT_PROVENANCE_CHECKER_UNHAUTHENTICATED = metrics.counter(
+    "thoth_user_api_cache_hit_rate",
+    "Thoth User API cache hit rate",
+    is_authenticated="False",
+    service="provenance-checker",
+    env=Configuration.THOTH_DEPLOYMENT_NAME,
+)
+
+_CACHE_HITS_METRICS = {
+    "adviser": {
+        "authenticated": _METRIC_CACHE_HIT_ADVISER_AUTHENTICATED,
+        "unauthenticated": _METRIC_CACHE_HIT_ADVISER_UNHAUTHENTICATED,
+    },
+    "provenance-checker": {
+        "authenticated": _METRIC_CACHE_HIT_PROVENANCE_CHECKER_AUTHENTICATED,
+        "unauthenticated": _METRIC_CACHE_HIT_PROVENANCE_CHECKER_UNHAUTHENTICATED,
```

Comment:
Would it make sense to move these authenticated/unauthenticated to metric labels? I guess it could case removal of these dictionaries and the metric can be directly set in the endpoint function handler (`_set_metrics_service_cache_hit` could be deleted).

Reply:
Removed all extra code and reduced to minimum if else condition. wdyt?